nidirect_breadcrumbs/src/PublicationBreadcrumb.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Drupal\nidirect_breadcrumbs;
+
+/**
+ * @file
+ * Generates the breadcrumb trail for Publication pages/content.
+ */
+
+use Drupal\Core\Breadcrumb\Breadcrumb;
+use Drupal\Core\Breadcrumb\BreadcrumbBuilderInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Link;
+use Drupal\Core\Url;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+class PublicationBreadcrumb implements BreadcrumbBuilderInterface {
+
+  /**
+   * Drupal entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Node object, or null if on a non-node page.
+   *
+   * @var \Drupal\node\Entity\Node
+   */
+  protected $node;
+
+  /**
+   * Class constructor.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager) {
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function applies(RouteMatchInterface $route_match) {
+    $match = FALSE;
+    $this->node = $route_match->getParameter('node');
+
+    if (!empty($this->node)) {
+      if (is_object($this->node) == FALSE) {
+        $this->node = $this->entityTypeManager->getStorage('node')->load($this->node);
+      }
+
+      $match = $this->node->bundle() == 'publication';
```

Comment:
What if we don't get a node returned from ETM and try calling bundle() on it, presumably it'll throw a method not found error.

Reply:
Good shout. I see the other handlers do the same, so I'll move this line into the block above and make this more defensive.

Reply:
Introduced a slightly clearer approach to this. I expect they'll evolve again as the project expands. I'd be inclined to move things like search page route names in as service parameters and maybe restructure the scope of the handlers a little but happy to tackle that once things become an issue rather than ahead of the time.